Dynatrace/dynatrace-operator
# Makes readonly oneagent host volume the default

# Description

- By default the oneagents get csi-volumes and readonly host mount in case of `cloudNative`/`hostMonitoring`.
  - `alpha.operator.dynatrace.com/feature-disable-oneagent-readonly-host-fs` can be used to disable it
- Updates the validation check that makes sure that the csi-driver is present. 
  - `hostMonitoring` needs it if readonly is not disabled
- Updates the template logic in helm to handle this new default and keep intact the "smart" way of deploying the csi-driver.
   - We only deploy the csi stuff in case the mode requires it. So now that `hostMonitoring` uses the csi driver by default, so some changes needed to be made.
- Adds checks during daemonset creation to make sure that classicFullStack daemonset wont be readonly (with opt in feature flag this was not super needed)

## How can this be tested?
Just deploy it with any mode(that deploys a oneagent daemonset) you want and it should just work.
In case of:
- classicFullStack: the oneagent daemonset should NOT have a csi volume and readonly host volume
- cloudNativeFullStack: the oneagent daemonset SHOULD have a csi volume and readonly host volume
- hostMonitoring: the oneagent daemonset SHOULD have a csi volume and readonly host volume

if the `alpha.operator.dynatrace.com/feature-disable-oneagent-readonly-host-fs: "true"` is present on the dynakube the oneagent daemonset should NOT have a csi volume and readonly host volume, no matter what.

if no csidriver is deployed, the validation webhook would deny hostMonitoring dynakubes (without the annotation)

## Checklist
- [x] Labeled the PR accordingly
